### PR TITLE
Some transactions do not mine when we use geth-binary and IPC provider

### PR DESCRIPTION
**Pull request by avkos**
## Description
#5730 
Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../blob/1.x/CONTRIBUTIONS.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth/test/integration/subscription_heads.test.ts [hunk 1]
```diff
@@ -15,38 +15,32 @@ You should have received a copy of the GNU Lesser General Public License
...
 import {
 	closeOpenConnection,
-	createTempAccount,
 	describeIf,
 	getSystemTestProvider,
 	isSocket,
+	sendFewSampleTxs,
 	waitForOpenConnection,
 } from '../fixtures/system_test_utils';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (luu-alex) [inline packages/web3-eth/test/integration/subscription_heads.test.ts]: think this is unneccesary

**Review by luu-alex — APPROVED** [review]

**Review by jdevcs — APPROVED** [review]

**Review by nikoulai — APPROVED** [review]

File: packages/web3-eth/test/integration/subscription_heads.test.ts (hunk 1 above)
Comment (avkos) [reply packages/web3-eth/test/integration/subscription_heads.test.ts]: done

File: packages/web3-eth-personal/test/integration/personal.test.ts [hunk 2]
```diff
@@ -106,6 +107,8 @@ describe('personal integration tests', () => {
 		const accountList = await ethPersonal.getAccounts();
 		// create a new account
 		const account = await ethPersonal.newAccount('cde');
+		// we need this for confirm transaction
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-personal/test/integration/personal.test.ts]: which transaction exactly that is here needs to be confimed?

File: packages/web3-eth-contract/test/integration/contract_deploy.test.ts [hunk 3]
```diff
@@ -148,7 +149,7 @@ describe('contract', () => {
 
 		it('should emit the "confirmation" event', async () => {
 			const confirmationHandler = jest.fn();
-
+			contract.setConfig({ transactionConfirmationBlocks: 2 });
```
Comment (Muhammad-Altabba) [inline packages/web3-eth-contract/test/integration/contract_deploy.test.ts]: This is good, I think. And would not the value `1`, if set to `transactionConfirmationBlocks` makes things even faster? 

```suggestion
			contract.setConfig({ transactionConfirmationBlocks: 1 });
```

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts [hunk 4]
```diff
@@ -19,62 +19,60 @@ import { DEFAULT_RETURN_FORMAT } from 'web3-utils';
 import { Web3PromiEvent } from 'web3-core';
 import { TransactionReceipt } from 'web3-types';
 import { TransactionBlockTimeoutError } from 'web3-errors';
-import { SendTransactionEvents, Web3Eth } from '../../src';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Web3 } from 'web3';
```
Comment (Muhammad-Altabba) [inline packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts:22]: I think it is better to keep using `Web3Eth` except if there is a strong reason for not doing so.

File: packages/web3-eth-personal/test/integration/personal.test.ts (hunk 2 above)
Comment (avkos) [reply packages/web3-eth-personal/test/integration/personal.test.ts]: yes. you are right. removed

File: packages/web3-eth-contract/test/integration/contract_deploy.test.ts (hunk 3 above)
Comment (avkos) [reply packages/web3-eth-contract/test/integration/contract_deploy.test.ts]: done

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts (hunk 4 above)
Comment (avkos) [reply packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts:22]: To use a local wallet, we need to use web3. because web3-eth doesn't have functionality to sign transactions from wallet

**Review by Muhammad-Altabba — APPROVED** [review]

File: packages/web3-eth-contract/test/integration/contract_deploy.test.ts [hunk 5]
```diff
@@ -36,14 +37,14 @@ describe('contract', () => {
 		let acc: { address: string; privateKey: string };
 		let pkAccount: { address: string; privateKey: string };
 		beforeAll(async () => {
-			pkAccount = await createNewAccount({ refill: true });
-			acc = await createTempAccount();
 			deployOptions = {
 				data: GreeterBytecode,
 				arguments: ['My Greeting'],
 			};
 		});
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/test/integration/contract_deploy.test.ts:44]: This entire `beforeAll` can be removed if `deployOptions` is set upon initialization

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts [hunk 6]
```diff
@@ -19,62 +19,60 @@ import { DEFAULT_RETURN_FORMAT } from 'web3-utils';
...
 describe('defaults', () => {
-	let web3Eth: Web3Eth;
-	let eth2: Web3Eth;
+	let web3: Web3;
 	let clientUrl: string;
+	let account1: Web3Account;
+	let account2: Web3Account;
 
-	beforeEach(() => {
+	beforeEach(async () => {
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts]:
```suggestion
	beforeEach(() => {
```

`await` is never used in this code block?

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts [hunk 7]
```diff
@@ -19,62 +19,60 @@ import { DEFAULT_RETURN_FORMAT } from 'web3-utils';
...
-				value,
-				from,
+			> = web3.eth.sendTransaction({
+				from: account1.address,
+				to: account2.address,
+				gas,
+				value: '0x1',
 				// Give a high nonce so the transaction stuck forever.
 				// However, make this random to be able to run the test many times without receiving an error that indicate submitting the same transaction twice.
 				nonce: Number.MAX_SAFE_INTEGER - Math.floor(Math.random() * 100000000),
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts]:
```suggestion
				nonce: Number.MAX_SAFE_INTEGER ,
```
This logic isn't need for this test and others if we're using a new account for each test?

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts [hunk 8]
```diff
@@ -102,43 +94,31 @@ describe('defaults', () => {
 				// eslint-disable-next-line jest/no-conditional-expect
 				expect((error as Error).message).toMatch(/was not mined within [0-9]+ blocks/);
 			}
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts:95]: These `try/catch`es should be refactored to look like [this](https://github.com/web3/web3.js/blob/82e9b03ca7396b63863761678c93dbf4dc0ed8a7/packages/web3-eth/test/integration/web3_eth/send_transaction.test.ts#L544-L548) so the test will fail on it's own if no error is received - also these checks should be testing the `error` event is emitted as expected

But this isn't directly related to the purpose of this PR, so an issue could be made to follow up on these changes

File: packages/web3-eth/test/integration/helper.ts [hunk 9]
```diff
@@ -15,95 +15,47 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 import { AbiEventFragment, Block, TransactionInfo, TransactionReceipt } from 'web3-types';
-import { DEFAULT_RETURN_FORMAT, FMT_NUMBER } from 'web3-utils';
-import { Web3PromiEvent } from 'web3-core';
-import { SendTransactionEvents, Web3Eth } from '../../src';
+import { FMT_NUMBER } from 'web3-utils';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/helper.ts:19]:
```suggestion
```

File: packages/web3-eth/test/integration/nonce.test.ts [hunk 10]
```diff
@@ -14,53 +14,54 @@ GNU Lesser General Public License for more details.
 You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
-// eslint-disable-next-line import/no-extraneous-dependencies
 import { TransactionPollingTimeoutError, TransactionSendTimeoutError } from 'web3-errors';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/nonce.test.ts:18]:
```suggestion
```

File: packages/web3-eth/test/integration/nonce.test.ts [hunk 11]
```diff
@@ -14,53 +14,54 @@ GNU Lesser General Public License for more details.
...
 import { Web3Eth } from '../../src';
 
 import {
 	closeOpenConnection,
-	createTempAccount,
+	createAccount,
+	createLocalAccount,
 	getSystemTestProvider,
 } from '../fixtures/system_test_utils';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/nonce.test.ts]:
```suggestion
```

File: packages/web3-eth/test/integration/subscription_new_pending_tx.test.ts [hunk 12]
```diff
@@ -14,42 +14,32 @@ GNU Lesser General Public License for more details.
 You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/subscription_new_pending_tx.test.ts:17]:
```suggestion
```

File: yarn.lock
Comment (Spacesai1or) [inline yarn.lock:1]: This change shouldn't be necessary if no packages were upgraded or added?

File: scripts/system_tests_utils.ts [hunk 13]
```diff
@@ -259,7 +259,7 @@ const walletsOnWorker = 20;
 if (tempAccountList.length === 0) {
 	tempAccountList = accountsString;
 }
-let currentIndex = 0;
+let currentIndex = Math.floor(Math.random() * tempAccountList.length);
```
Comment (Spacesai1or) [inline scripts/system_tests_utils.ts:262]: What's the purpose of this change? Why not start at 0?

File: packages/web3-eth/test/integration/watch_transaction.test.ts [hunk 14]
```diff
@@ -17,43 +17,52 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 import { DEFAULT_RETURN_FORMAT } from 'web3-utils';
 import { TransactionReceipt } from 'web3-types';
 import { Web3PromiEvent } from 'web3-core';
-import { Web3Eth, SendTransactionEvents } from '../../src';
-import { sendFewTxesWithoutReceipt } from './helper';
-
+import { Web3Account } from 'web3-eth-accounts';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/watch_transaction.test.ts:21]:
```suggestion
```

File: packages/web3-eth/test/integration/subscription_heads.test.ts [hunk 15]
```diff
@@ -15,38 +15,31 @@ You should have received a copy of the GNU Lesser General Public License
 along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 */
 import { BlockHeaderOutput } from 'web3-types';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/subscription_heads.test.ts:18]:
```suggestion
```

File: packages/web3-eth/test/integration/subscription_heads.test.ts [hunk 16]
```diff
@@ -15,38 +15,31 @@ You should have received a copy of the GNU Lesser General Public License
...
-	let tempAcc2: { address: string; privateKey: string };
-
-	beforeEach(async () => {
-		tempAcc2 = await createTempAccount();
-	});
-	beforeAll(() => {
+	let web3: Web3;
+	beforeAll(async () => {
 		clientUrl = getSystemTestProvider();
 	});
```
Comment (Spacesai1or) [inline packages/web3-eth/test/integration/subscription_heads.test.ts]: This `beforeAll` doesn't need to be here, can assign `clientUrl` upon initialization

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts (hunk 7 above)
Comment (avkos) [reply packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts:78]: it is was done before me

File: packages/web3-eth/test/integration/helper.ts (hunk 9 above)
Comment (avkos) [reply packages/web3-eth/test/integration/helper.ts:19]: we need this to import web3

File: packages/web3-eth/test/integration/subscription_new_pending_tx.test.ts (hunk 12 above)
Comment (avkos) [reply packages/web3-eth/test/integration/subscription_new_pending_tx.test.ts:17]: need to import web3

File: scripts/system_tests_utils.ts (hunk 13 above)
Comment (avkos) [reply scripts/system_tests_utils.ts:262]: Tests for every package start with 0 and gets accounts from a list. So, to make accounts equal in use, I've added a random start index

File: packages/web3-eth/test/integration/nonce.test.ts (hunk 11 above)
Comment (avkos) [reply packages/web3-eth/test/integration/nonce.test.ts]: done

File: packages/web3-eth/test/integration/nonce.test.ts (hunk 10 above)
Comment (avkos) [reply packages/web3-eth/test/integration/nonce.test.ts:18]: done

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts (hunk 7 above)
Comment (avkos) [reply packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts]: done

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts (hunk 6 above)
Comment (avkos) [reply packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts]: done

File: packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts (hunk 8 above)
Comment (avkos) [reply packages/web3-eth/test/integration/defaults.transactionBlockTimeout.test.ts:96]: done

File: packages/web3-eth/test/integration/watch_transaction.test.ts (hunk 14 above)
Comment (avkos) [reply packages/web3-eth/test/integration/watch_transaction.test.ts:21]: done

File: packages/web3-eth/test/integration/subscription_heads.test.ts (hunk 15 above)
Comment (avkos) [reply packages/web3-eth/test/integration/subscription_heads.test.ts:18]: this one we need to import web3

File: packages/web3-eth/test/integration/subscription_heads.test.ts (hunk 16 above)
Comment (avkos) [reply packages/web3-eth/test/integration/subscription_heads.test.ts]: done

File: yarn.lock
Comment (avkos) [reply yarn.lock:1]: done

File: packages/web3-eth-contract/test/integration/contract_deploy.test.ts (hunk 5 above)
Comment (avkos) [reply packages/web3-eth-contract/test/integration/contract_deploy.test.ts:44]: done
